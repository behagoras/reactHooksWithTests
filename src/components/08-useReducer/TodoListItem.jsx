import React from 'react'

export default function TodoListItem({ todo = {}, editTodo, deleteTodo }) {

  return (
    <li key={todo.id} className="list-group-item">
      <input
        type="checkbox"
        name="done"
        id="done"
        checked={todo.done}
        className="form-check-input"
        onChange={(e) => editTodo({ id: todo.id, done: e.target.checked })}
      />
      <input
        type="text"
        name="todo"
        id="todo"
        value={todo.todo}
        className={`form-control todo ${todo.done ? 'complete' : ''}`}
        placeholder="Add your todo description"
        onChange={(e) => editTodo({ id: todo.id, todo: e.target.value })}
      />
      <div className="col-5">
        <button type="button" className="btn btn-outline-primary" onClick={() => deleteTodo(todo.id)}>-</button>
      </div>
    </li>
  )
}
