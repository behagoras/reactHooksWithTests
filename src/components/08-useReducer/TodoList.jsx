import React from 'react'
import TodoListItem from './TodoListItem'

export default function TodoList({ todos = [], editTodo, deleteTodo }) {

  return (
    <ul className="list-group list-group-flush col-7">
      {todos.map((todo) => (<TodoListItem todo={todo} key={todo.id} editTodo={editTodo} deleteTodo={deleteTodo} />))}
    </ul>
  )
}
