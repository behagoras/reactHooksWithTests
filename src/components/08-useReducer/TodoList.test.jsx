import React from 'react'
import { shallow } from 'enzyme'
import TodoList from './TodoList'
import { todosFixture } from '../../fixtures/todoFixtures'
import TodoListItem from './TodoListItem'

describe('<TodoList />', () => {
  let component = shallow(<TodoList />)
  const editTodo = jest.fn()
  const deleteTodo = jest.fn()
  it('should render correctly', () => {
    component = shallow(<TodoList
      todos={todosFixture}
      editTodo={editTodo}
      deleteTodo={deleteTodo}
    />)
    expect(component).toMatchSnapshot()
  })
  it('should render the correct quantity of elements', () => {
    const toDoListItem = component.find(TodoListItem)
    const firstTodo = toDoListItem.at(0)

    expect(toDoListItem.length).toBe(todosFixture.length)

    expect(firstTodo.prop('deleteTodo')).toBe(deleteTodo)
    expect(firstTodo.prop('editTodo')).toBe(editTodo)
  })
})
