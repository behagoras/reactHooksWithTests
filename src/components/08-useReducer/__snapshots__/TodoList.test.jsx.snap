// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render correctly 1`] = `
<ul
  className="list-group list-group-flush col-7"
>
  <TodoListItem
    deleteTodo={[MockFunction]}
    editTodo={[MockFunction]}
    key="123"
    todo={
      Object {
        "done": true,
        "id": "123",
        "todo": "read a book",
      }
    }
  />
  <TodoListItem
    deleteTodo={[MockFunction]}
    editTodo={[MockFunction]}
    key="234"
    todo={
      Object {
        "done": false,
        "id": "234",
        "todo": "learn new tech",
      }
    }
  />
</ul>
`;
